app/models/apple/config.rb

```diff
@@ -99,18 +90,21 @@ def self.setup_delegated_delivery(podcast, key: nil, apple_config: nil, apple_sh
       mark_as_delivered!(pub)
     end
 
-    def feed_podcasts_match
-      return unless public_feed.present? && private_feed.present?
+    def not_default_feed
+      if feed&.default?
+        errors.add(:feed, "cannot use default feed")
+      end
+    end
 
-      if (public_feed.podcast_id != podcast_id) || (private_feed.podcast_id != podcast_id)
-        errors.add(:public_feed, "must belong to the same podcast as the private feed")
+    def podcast_has_one_apple_config
+      all_feeds = Feed.where(podcast_id: feed.podcast_id).pluck(:id)
+      if Apple::Config.where(feed_id: all_feeds).where.not(id: id).any?
+        errors.add(:feed, "podcast already has an apple config")
       end
     end
 
     def publish_to_apple?
-      return false unless key&.valid?
-
-      public_feed.publish_to_apple?
+      !!key&.valid? && publish_enabled?
```

Comment:
The logic was circular here - it called up to the `public_feed`, which called back down into this `apple_config.publish_enabled` field.  Cleaned it up so this model gets to decide directly.

Reply:
I saw this, and almost changed it in my PR, but decided it was a bit out of scope - glad you saw it too!